src/shim/umq/hwq.cpp

```diff
@@ -202,6 +216,10 @@ fill_slot_and_send(volatile host_queue_packet_t *pkt, void *payload, size_t size
   std::memcpy(data, payload, size);
   /* Always done as last step. */
   mark_slot_valid(pkt);
+
+  /* flash cache line for non coherence memory */
+  pkt_clflush(pkt);
```

Comment:
Here you do flush only for the valid bit change, so no need to flush the whole pkg.

Reply:
If we only flush the valid "uint16_t type: 8;", the memcpy(data, ...) could be still not flushed right? So the device can get dirty data?

Flushing the valid bit is easier than flush the whole data. 

I think we'd better have a chat on this.

Reply:
@xdavidz If clflush is for both data and valid bit, let's assume valid bit is flushed to memory but data is still not fully flushed, will the device start access data because of valid bit is set? Is there any race in this case?

Reply:
> @xdavidz If clflush is for both data and valid bit, let's assume valid bit is flushed to memory but data is still not fully flushed, will the device start access data because of valid bit is set? Is there any race in this case?

I think we could should make sure the date is flushed first;
then we start writing the valid bit second;
finally we should flush the valid bit again;

I will discuss this flow with Max when he comes back online. Otherwise, there is window that data is in the cache but not in the memory.